### PR TITLE
Fix conditions of "stop on error" and "continue on error" in E2E tests 

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/630
Regression: Yes  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: Exchange the conditions of "stop on error" and "continue on error". 

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  
Validation:

### CODE REVIEW

**Review by nkolev92 — APPROVED** [review]

File: build/templates/End_To_End_Tests_On_Windows.yml [hunk 1]
```diff
@@ -67,7 +67,7 @@ steps:
   inputs:
     scriptName: "$(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\scripts\\RunFunctionalTests.ps1"
     arguments: "-PMCCommand $(EndToEndTestCommandToRunPart) -PMCLaunchWaitTimeInSecs 30 -EachTestTimoutInSecs 600 -NuGetDropPath $(System.DefaultWorkingDirectory)\\artifacts -FuncTestRoot $(System.DefaultWorkingDirectory)\\artifacts -RunCounter $(Build.BuildNumber) -VSVersion 16.0"
-  condition: "and(succeeded(), eq(variables['IsOfficialBuild'], 'true'))"
+  condition: "and(succeeded(), not(eq(variables['IsOfficialBuild'], 'true')))"
```
Comment (zivkan) [inline build/templates/End_To_End_Tests_On_Windows.yml:70]: the `continueOnError: "true"` on line 66 needs to be deleted.

Comment (heng-liu) [reply build/templates/End_To_End_Tests_On_Windows.yml:70]: Thanks. Changed continueOnError to "false" in line 66.

**Review by zivkan — APPROVED** [review]
Line 66 could be deleted entirely, rather than set to false. But either way this looks right to me.

**Review by kartheekp-ms — APPROVED** [review]
+1 to Andy's comment because `continueOnError` defaults to [false](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/tasks?view=azure-devops&tabs=yaml#task-control-options).

**Comment by heng-liu** [comment]
I'll keep as it is, because I think setting `continueOnError `to false is more clear. And it's consistent with other "stop on error" tasks, like https://github.com/NuGet/NuGet.Client/blob/dev/build/vsts_build.yaml#L209

**Pull request merged**